### DISCUSSION TITLE
authWithPassword and MFA _superuser

### DISCUSSION THREAD

**Discussion by cem-al** [pocketbase/pocketbase#6032] [Q&A]
If MFA is enabled for `_superusers`, how do I authenticate a pocketbase instance where I would usually use username and password? eg. in unit tests, or where user context is not available.

For instance, this is how I currently do it:

```js
const authResult = await pb.admins.authWithPassword(
   process.env.POCKET_BASE_ADMIN_USER,
   process.env.POCKET_BASE_ADMIN_PASS,
);
```

But with MFA enabled this fails.

**Comment by ganigeorgiev** [comment]
Why do you expect not to fail? If you enabled MFA for the `_superusers` collection then it will be required. The behavior is the same for the regular auth collections.

I don't exactly understand what do you mean by "_in unit tests, or where user context is not available_" (how do you write unit tests with the JS SDK?) but for the test environment you can simply disable it, assuming that you have a separate test `pb_data`.

Another alternative could be to use "API keys" based auth - https://pocketbase.io/docs/authentication/#api-keys.

**Reply by ganigeorgiev** [reply]
_Also note that `pb.admins.*` is soft-deprecated and it is recommend to replace it with `pb.collection("_superusers").*` (see the [JS SDK v0.22.0 CHANGELOG](https://github.com/pocketbase/js-sdk/releases/tag/v0.22.0))._

**Reply by chnfyi** [reply]
@ganigeorgiev i ping you here exceptionnally instead of opening an issue, to notice you that _Auth with MFA_ is missing in the UI API preview.

**Reply by ganigeorgiev** [reply]
I don't think it is necessary. There is already a "Learn more" link in the collection options (_I may consider eventually adding the link in the description of the auth methods preview if we detect that the collection has the MFA option enabled_).

**Comment by chnfyi** [comment]
Make sure you enabled the One-time password (OTP) auth collection option too.

And you'll need to use your superuser email anyway.

```javascript
try {
  await pb.collection('_superusers').authWithPassword(
    process.env.POCKET_BASE_ADMIN_USER, 
    process.env.POCKET_BASE_ADMIN_PASS
)} catch (err) {
  const mfaId = err.response?.mfaId;
  if (!mfaId) {
    throw err; // not mfa -> rethrow
  }

  const result = await pb.collection('_superusers').requestOTP(process.env.POCKET_BASE_ADMIN_EMAIL);
  await pb.collection('_superusers').authWithOTP(result.otpId, 'EMAIL_CODE', { 'mfaId': mfaId });
}
```
